packages/package_info/README.md

```diff
@@ -5,12 +5,15 @@ application package.
 
 # Usage
 
-You can use the `PackageInfo` class to query information about the
+You can use the package_info to query information about the
 application package. This works both on iOS and Android.
 
 ```dart
-var version = await PackageInfo.getVersion();
-var buildNumber = await PackageInfo.getBuildNumber();
+import 'package:package_info/package_info.dart' as package_info;
+
+var version = await package_info.version;
+var buildNumber = await package_info.buildNumber;
+var packageName = await package_info.packageName;
```

Comment:
Would it make sense to refactor this into the design used by `shared_preferences` (https://github.com/flutter/plugins/blob/master/packages/shared_preferences/lib/shared_preferences.dart), so that we'd have a single await to get all package info in one go?

Reply:
I can do this in next pull request.

Reply:
@mravn-google How about refactor below:

```
/// Cache infomation [Map] about the current application package.
///
class PackageInfo {
  PackageInfo._({this.version, this.buildNumber, this.packageName});

  static PackageInfo _instance;

  static Future<PackageInfo> getInstance() async {
    if (_instance == null) {
      final Map<String, String> fromSystem = await _kChannel.invokeMethod('getAll');
      assert(fromSystem != null);
      assert(fromSystem["version"] != null);
      assert(fromSystem["buildNumber"] != null);
      assert(fromSystem["packageName"] != null);

      _instance = new PackageInfo._(
        version: fromSystem["version"],
        buildNumber: fromSystem["buildNumber"],
        packageName: fromSystem["packageName"],
      );
    }
    return _instance;
  }

  /// Returns the `CFBundleShortVersionString` on iOS or `versionName` on Android
  final String version;

  /// Returns the `CFBundleVersion` on iOS or `versionCode` on Android
  final String buildNumber;

  /// Returns the `bundleIdentifier` on iOS or `getPackageName` on Android
  final String packageName;
}

/// Returns the `CFBundleShortVersionString` on iOS or `versionName` on Android
/// this method is now deprecated, use [PackageInfo.version] property instead.
@deprecated
Future<String> get version async => await _kChannel.invokeMethod('getVersion');

/// Returns the `CFBundleVersion` on iOS or `versionCode` on Android
/// this method is now deprecated, use [PackageInfo.buildNumber] property instead.
@deprecated
Future<String> get buildNumber async => await _kChannel.invokeMethod('getBuildNumber');

/// Returns the `bundleIdentifier` on iOS or `getPackageName` on Android
/// this method is now deprecated, use [PackageInfo.packageName] property instead.
@deprecated
Future<String> get packageName async => await _kChannel.invokeMethod('getPackageName');
```

Reply:
- The dartdoc on `PackageInfo` should describe what it is (information about the application package), not how it is implemented (via a `Map`).
- If we are sure that the three pieces of information cannot be null on the platform side, the three constructor parameters should be marked `@required`.
- The constructor's initializer list can do the `assert xxx != null`.
- The dartdoc on the final fields should not start with "Returns ".
- I don't think it makes sense to deprecate the old async methods. Just remove them, add a bold face Breaking change entry to the CHANGELOG.md, and bump the version number to `0.2.0`. After all, consuming apps are in control of when to upgrade their plugin dependencies.

It's harder to provide review feedback this way. Please create the PR, and we'll take it from there.